.github/workflows/uncrustify.yml

```diff
@@ -0,0 +1,49 @@
+name: Uncrustify the source code
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  Uncrustify:
+    name: Run_Uncrustify
+    if: ${{ github.event.issue.pull_request && github.event.comment.body == '/bot run uncrustify' }}
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+      - name: get pullrequest url
+        run: |
+          echo ${{ github.event.issue.pull_request.url }}
+      - name: get upstream repo
+        id: upstreamrepo
+        run: |
+          echo "::set-output name=RemoteRepo::$(curl -H "Accept: application/vnd.github.sailor-v-preview+json" --url ${{ github.event.issue.pull_request.url }} | jq '.head.repo.full_name' | sed 's/\"//g')"
+      - name: get upstream branch
+        id: upstreambranch
+        run: |
+          echo "::set-output name=branchname::$(curl -H "Accept: application/vnd.github.sailor-v-preview+json" --url ${{ github.event.issue.pull_request.url }} | jq '.head.ref' | sed 's/\"//g')"
+      - name: echo upstream repo:branch
+        run: |
+          echo ${{ steps.upstreamrepo.outputs.RemoteRepo }}:${{ steps.upstreambranch.outputs.branchname }}
+      - name: Checkout upstream repo
+        uses: actions/checkout@v2
+        with:
+           token: ${{ secrets.PAT }}
+           repository: ${{ steps.upstreamrepo.outputs.RemoteRepo }}
+           ref: ${{ steps.upstreambranch.outputs.branchname }}
+      - name: Install Uncrustify
+        run: sudo apt-get install uncrustify
```

Comment:
@cobusve mentioned that there have been compatibility issues between different versions of uncrustify and our config. It might be best to pin this to a particular version.

Reply:
Yes, as I mentioned above, Ubuntu 18.04 comes with a certain version of uncrustify. And that is what is suitable for us. If that changes, we'd need to update this too.